### PR TITLE
[NMS] Refresh equipment and subscriber detail

### LINKED ISSUE #4405: [NMS] Fix Subscriber Add/Edit unit test

### PR DESCRIPTION
**Pull request by HannaFar**
Signed-off-by: HannaFar <hannafarag159@gmail.com>

## Summary

Refresh gateway detail enodeb and subscriber table.
Add autoRefresh checkbox to enodeb, gateway and subscriber detail.
Fix subscriber add/edit unit test.
Close #4405 

Capture d’écran 2021-01-20 à 15 39 49

Capture d’écran 2021-01-20 à 15 39 22

Capture d’écran 2021-01-20 à 15 30 16

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailMain.js [hunk 1]
```diff
@@ -360,27 +364,47 @@ function GatewayOverview() {
...
     </div>
   );
 }
 
+function GatewayInfo() {
+  const {match} = useRouter();
+  const networkId: string = nullthrows(match.params.networkId);
+  const gatewayId: string = nullthrows(match.params.gatewayId);
+  // Auto refresh gateways every 30 seconds
+  const state = useRefreshingContext({
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailMain.js]: We seem to be missing the autoRefresh checkbox. Can you also add that in here. 
We also have to add refresh functionality for GatewayDetailEnodebs(for refreshing eNodeB state) and GatewayDetailSubscribers(subscriber state), We want to do both to be default, not autorefreshing. Did you plan to make that as a separate PR ?

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailSubscribers.js [hunk 2]
```diff
@@ -28,23 +33,29 @@ type SubscriberRowType = {
...
   const subscriberCtx = useContext(SubscriberContext);
   const gwSubscriberMap =
-    subscriberCtx.gwSubscriberMap[gwInfo?.device?.hardware_id] || [];
+    subscriberCtx.gwSubscriberMap[props.gwInfo?.device?.hardware_id] || [];
 
   const subscriberRows: Array<SubscriberRowType> = gwSubscriberMap.map(
     (serialNum: string) => {
-      const subscriberInfo = subscriberCtx.state[serialNum];
+      // $FlowIgnore
+      const subscriberInfo = subscriberState.state?.[serialNum];
```
Comment (andreilee) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailSubscribers.js:54]: What's the flow error that comes up here?

File: nms/app/packages/magmalte/app/components/context/RefreshContext.js [hunk 3]
```diff
@@ -95,6 +95,12 @@ export function useRefreshingContext(props: Props) {
     return ctx.setState(null, null, newState);
   }
 
+  // Avoid using state as a dependency of useEffect
+  const stateRef = useRef(null);
+  useEffect(() => {
+    stateRef.current = state;
+  });
```
Comment (andreilee) [inline nms/app/packages/magmalte/app/components/context/RefreshContext.js]:
```suggestion
  useEffect(() => {
    stateRef.current = state;
  }, [state]);
```

Should this only trigger on `state` updating?

**Review by andreilee — APPROVED** [review]

File: nms/app/packages/magmalte/app/components/context/RefreshContext.js (hunk 3 above)
Comment (HannaFar) [reply nms/app/packages/magmalte/app/components/context/RefreshContext.js]: Yes you're right!

Comment (HannaFar) [reply nms/app/packages/magmalte/app/components/context/RefreshContext.js]: Yes!

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailSubscribers.js (hunk 2 above)
Comment (HannaFar) [reply nms/app/packages/magmalte/app/views/equipment/GatewayDetailSubscribers.js:54]: Cannot get `subscriberState.state` because property `state` is missing in object literal

File: nms/app/packages/magmalte/app/views/equipment/GatewayDetailMain.js [hunk 4]
```diff
@@ -339,6 +345,18 @@ function GatewayOverview() {
   const gatewayId: string = nullthrows(match.params.gatewayId);
   const gwCtx = useContext(GatewayContext);
   const gwInfo = gwCtx.state[gatewayId];
+  const [refresh, setRefresh] = useState(false);
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/equipment/GatewayDetailMain.js]: minor nit: set the default refresh for gateways as true. For eNodeBs and subscribers, setting false is fine.

File: nms/app/packages/magmalte/app/views/equipment/EnodebDetailMain.js [hunk 5]
```diff
@@ -166,6 +168,7 @@ function Overview() {
   const classes = useStyles();
   const [startDate, setStartDate] = useState(moment().subtract(3, 'hours'));
   const [endDate, setEndDate] = useState(moment());
+  const [refresh, setRefresh] = useState(false);
```
Comment (karthiksubraveti) [inline nms/app/packages/magmalte/app/views/equipment/EnodebDetailMain.js]: Let's keep the default to true

**Review by karthiksubraveti — APPROVED** [review]
Everything else LGTM. Left a few comments.
